RecoPixelVertexing/PixelTriplets/plugins/CAHitQuadrupletGenerator.cc

```diff
@@ -216,21 +247,194 @@ void CAHitQuadrupletGenerator::hitNtuplets(const IntermediateHitDoublets::Region
+    		{
+      			return id == PixelSubdetector::PixelBarrel;
+    		};
+    		for(unsigned int i = 0; i< 3; ++i)
+    		{
+        		auto const& ahit = foundQuadruplets[quadId][i]->getInnerHit();
+        		gps[i] = ahit->globalPosition();
+        		ges[i] = ahit->globalPositionError();
+        		barrels[i] = isBarrel(ahit->geographicalId().subdetId());
+    		}
+
+    		auto const& ahit = foundQuadruplets[quadId][2]->getOuterHit();
+    		gps[3] = ahit->globalPosition();
+    		ges[3] = ahit->globalPositionError();
+    		barrels[3] = isBarrel(ahit->geographicalId().subdetId());
+    		if(caOnlyOneLastHitPerLayerFilter)
+   		{
+ 	     		const auto fourthLayerId = tTopo->layer(ahit->geographicalId());
+            		const auto sideId = tTopo->side(ahit->geographicalId());
+            		const auto subDetId = ahit->geographicalId().subdetId();
+    			const auto isTheSameTriplet = (quadId != 0) && (foundQuadruplets[quadId][0]->getCellId() ==  previousCellIds[0]) && (foundQuadruplets[quadId][1]->getCellId() ==  previousCellIds[1]);
+   			const auto isTheSameFourthLayer = (quadId != 0) &&  (fourthLayerId == previousfourthLayerId) && (subDetId == previousSubDetId) && (sideId == previousSideId);
+
+    			previousCellIds = {{foundQuadruplets[quadId][0]->getCellId(), foundQuadruplets[quadId][1]->getCellId()}};
+    			previousfourthLayerId = fourthLayerId;
+
+
+    			if(!(isTheSameTriplet && isTheSameFourthLayer ))
+    			{
+    				selectedChi2 = std::numeric_limits<float>::max();
+    				hasAlreadyPushedACandidate = false;
+    			}
+
+    		}
+    		// TODO:
+    		// - 'line' is not used for anything
+    		// - if we decide to always do the circle fit for 4 hits, we don't
+    		//   need ThirdHitPredictionFromCircle for the curvature; then we
+    		//   could remove extraHitRPhitolerance configuration parameter
+    		PixelRecoLineRZ line(gps[0], gps[2]);
+    		ThirdHitPredictionFromCircle predictionRPhi(gps[0], gps[2], extraHitRPhitolerance);
+    		const float curvature = predictionRPhi.curvature(ThirdHitPredictionFromCircle::Vector2D(gps[1].x(), gps[1].y()));
+    		const float abscurv = std::abs(curvature);
+    		const float thisMaxChi2 = maxChi2Eval.value(abscurv);
+    		if (theComparitor)
+    		{
+      			SeedingHitSet tmpTriplet(foundQuadruplets[quadId][0]->getInnerHit(), foundQuadruplets[quadId][2]->getInnerHit(), foundQuadruplets[quadId][2]->getOuterHit());
+
+
+      			if (!theComparitor->compatible(tmpTriplet) )
+      			{
+       	 			continue;
+      			}
+    		}
+
+    		float chi2 = std::numeric_limits<float>::quiet_NaN();
+    		// TODO: Do we have any use case to not use bending correction?
+    		if (useBendingCorrection)
+   		{
+      		// Following PixelFitterByConformalMappingAndLine
+      		const float simpleCot = ( gps.back().z() - gps.front().z() ) / (gps.back().perp() - gps.front().perp() );
+     		const float pt = 1.f / PixelRecoUtilities::inversePt(abscurv, es);
```

Comment:
some indentation to offset the code in the ```if (useBendingCorrection)``` scope would be useful for readability